mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java

```diff
@@ -138,7 +138,9 @@ public boolean isChildOf(Mailbox potentialParent, MailboxSession mailboxSession)
     public boolean equals(Object obj) {
         if (obj instanceof Mailbox) {
             Mailbox o = (Mailbox)obj;
-            return Objects.equals(id, o.getMailboxId());
+            return Objects.equals(id, o.getMailboxId())
+                && Objects.equals(path, o.path)
+                && Objects.equals(uidValidity, o.uidValidity);
```

Comment:
Danger zone.

Please do not alter core POJOs, unless you really know what you do.

Reply:
reverted change